### PR TITLE
Out-of-the-box color picker

### PR DESCRIPTION
**Pull request by michelkaporin**
On-going work on #28587

### CODE REVIEW

File: extensions/css/client/src/cssMain.ts [hunk 1]
```diff
@@ -59,6 +60,18 @@ export function activate(context: ExtensionContext) {
 		};
 		disposable = activateColorDecorations(colorRequestor, { css: true, scss: true, less: true }, isDecoratorEnabled);
 		context.subscriptions.push(disposable);
+
+		languages.registerColorProvider(documentSelector, <DocumentColorProvider>{
+			provideDocumentColors(document: TextDocument, token: CancellationToken): ProviderResult<ColorInfo[]> {
+				const colorInfos: ColorInfo[] = [];
+				colorRequestor(document.uri.toString()).then(ranges => {
+					ranges.forEach(range => {
+						colorInfos.push(new ColorInfo(undefined, undefined, '', []));
```
Comment (michelkaporin) [inline extensions/css/client/src/cssMain.ts]: ping @joaomoreno for colour text parsing, thanks for help

**Review by joaomoreno — APPROVED** [review]
@michelkaporin I did a quick final review. The changes look good, so feel free to merge to `master` when you are ready, ideally before the next build kicks in. I've also added myself as a tester and will give it a good spin tomorrow.

Great work! I'm sure we can get it out for July! I will carry your legacy going forward with pride! 👍

**Comment by michelkaporin** [comment]
@joaomoreno happy to hear that! Thanks again for the help 🙂  I know it will be in your safe hands once I leave 👍 I will assign all issues that are left to you as agreed 💘

**Comment by vvs** [comment]
Hi, it looks like the color picker doesn't work inside SCSS files. I can only see the color picker in CSS ones.
Is this intentional?

File: src/vs/editor/common/modes.ts [hunk 2]
```diff
@@ -652,6 +652,76 @@ export interface LinkProvider {
...
+	format: IColorFormat;
+
+	/**
+	 * The available formats for this specific color.
+	 */
+	availableFormats: IColorFormat[];
+}
+
+/**
+ * A provider of colors for editor models.
```
Comment (rebornix) [inline src/vs/editor/common/modes.ts]: I love it's internal.

**Comment by michelkaporin** [comment]
@vvs addressing it in #31753

**Pull request merged**